test/reducers/user.spec.js

```diff
@@ -0,0 +1,22 @@
+import { expect } from 'chai';
+import reducer from '../../app/reducers/user';
+
+describe('reducers', () => {
+
+  describe('user', () => {    
+    
+    it('should handle USER_LOGIN', () => {
+      const action = { 
+        type: 'USER_LOGIN',
+        payload: {
+          username: 'John Doe', 
+          loggedIn: true 
+        }
+      };
+      const test = Object.assign({}, action.payload);
+      expect(reducer({}, action)).to.deep.equal(test);
```

Comment:
Since reducers return the whole state tree, would the deep equal fail when more actions are added?

Wondering if it should it be:
```js
const state = reducer({}, action)
expect(state).to.have.property('username', actions.payload.username)
expect(state).to.have.property('loggedIn', actions.payload.loggedIn)
```

Disclaimer: I haven't really written tests against reducers or components.